llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py

```diff
@@ -261,6 +280,36 @@ async def _aembed(self, texts: List[str], input_type: str) -> List[List[float]]:
             ).embeddings
         return getattr(result, self.embedding_type, None)
 
+    def _embed_image(self, image_path: str, input_type: str) -> List[float]:
+        """Embed images using Cohere."""
+        if self.model_name not in V3_MODELS:
+            raise ValueError(
+                f"{self.model_name} is not a valid multi-modal embedding model. Supported models are {MULTIMODAL_MODELS}"
+            )
+        client = self._get_client()
+        processed_image = self._image_to_base64_data_url(image_path)
+        return client.embed(
+            model=self.model_name,
+            images=[processed_image],
+            input_type=input_type,
+        ).embeddings
+
+    async def _aembed_image(self, image_path: str, input_type: str) -> List[float]:
```

Comment:
This only embeds a single image at a time -- do we want to support batches? By default, `_aget_image_embeddings` and `_get_image_embeddings` will call embeddings one at a time, but we could override to do proper batch calls

Reply:
currently does not support batches as discussed so not including it as of now.